src/lib/link.spec.ts

```diff
@@ -20,6 +20,47 @@ test.serial('establish new client-connection', async (t) => {
   // TODO: ensure it is updated
 });
 
+test.serial('reuse existing connections', async (t) => {
+  const [src, dest] = await setup();
+
+  const oldLink = await Link.createWithNewConnections(src, dest);
+  const connA = oldLink.endA.connectionID;
+  const connB = oldLink.endB.connectionID;
+
+  const newLink = await Link.createWithExistingConnections(
+    src,
+    dest,
+    connA,
+    connB
+  );
+
+  // ensure the data makes sense (TODO: more?)
+  t.assert(
+    newLink.endA.clientID.startsWith('07-tendermint-'),
+    newLink.endA.clientID
+  );
+  t.assert(
+    newLink.endB.clientID.startsWith('07-tendermint-'),
+    newLink.endB.clientID
+  );
+
+  // try to update both clients, ensuring this connection is stable
+  await newLink.updateClient('A');
+  // TODO: ensure it is updated
+  await newLink.updateClient('B');
+  // TODO: ensure it is updated
+});
+
+test.serial('errors when reusing an invalid connection', async (t) => {
+  const [src, dest] = await setup();
+
+  const connA = 'whatever';
+  const connB = 'unreal';
```

Comment:
This is a good example. I would propose 2 more.

1. Try to connect nodeA with nodeA using proper connectionid. Each side will pass local checks, but fail if we compare the client with the remote chain

Reply:
What's the second one?

Reply:
Ah, second one would be a partially open channel. Manually do init/try but don't finish the handshake (copy code from the whole flow in link).

Then give valid connection ids and valid clients, but it is in a non-open state (init or tryinit). You should ensure the connection is usable at startup